### DISCUSSION TITLE
How to evaluate an expression in the CLI

### DISCUSSION THREAD

**Discussion by odenix** [apple/pkl#707] [Q&A] [answered]
I tried `pkl eval -x "1 + 2"`, but it complains that no module is specified.
Instead, I had to do `pkl eval -x "1 + 2" EmptyFile.pkl`.
Is there a way to evaluate just an expression?
I couldn’t find an example for `-x` in the docs.

**Comment by HT154** [comment, marked as answer]
You're right that there's no way to just evaluate an expression in isolation (similar to `jq -n`). I found myself running `pkl eval -x '...' /dev/null` on occasion when I needed that so I ended up defining a shell alias:
```console
$ alias pklx='pkl eval /dev/null -x'
$ pklx '1 + 2'
3
```

There's some discussion and example for `-x` here: https://pkl-lang.org/main/current/pkl-cli/index.html#options

**Reply by odenix** [reply]
Thanks! I had missed that the docs can be expanded. I’ll file an issue for evaluating an expression in isolation.

**Reply by StefMa** [reply]
Maybe we can think about something better and build a full pkl REPL? 🤔

**Reply by HT154** [reply]
```
pkl repl
```

😉

**Reply by odenix** [reply]
Next stop: LSP based autocompletion for the REPL. :-)

**Reply by bioball** [reply]
For a command that works on all platforms:

```
pkl eval pkl:base -x '<expr>'
```

**Reply by bioball** [reply]
> Next stop: LSP based autocompletion for the REPL. :-)

Yeah, this is much needed. Would also be great to have syntax highlighting in there.
